zeroconf/__init__.py

```diff
@@ -651,15 +659,16 @@ def write(self, out):
         out.write_short(self.port)
         out.write_name(self.server)
 
+    @profile
     def __eq__(self, other):
         """Tests equality on priority, weight, port and server"""
         return (
             isinstance(other, DNSService)
-            and DNSEntry.__eq__(self, other)
             and self.priority == other.priority
             and self.weight == other.weight
             and self.port == other.port
             and self.server == other.server
+            and DNSEntry.__eq__(self, other)
```

Comment:
I ran an experiment to test the effectiveness of this change, measuring performance in a realistic situation.

I ran a performance test on a network with 170 devices running Zeroconf. On my computer there was a ServiceBrowser running on a separate thread while a timer ran on the main thread that forced a thread switch every 2 seconds (to include the effect of thread switching in the measurements). Every minute, a Zeroconf broadcast was made on the network. I ran this for an hour on a Macbook Air from 2015 (Intel Core i7-5650U) using Ubuntu 19.10 and Python 3.7, both before this commit and after.

These are the results of the performance tests:

| Function | Before count | Before time | Before time per count | After count | After time | After time per count | Time reduction |
|----|----|----|----|----|----|----|----|
|`DNSEntry.__eq__`|528|0.001s|1.9μs|538|0.001s|1.9μs|1.9%|
|`DNSPointer.__eq__`|24369256 (24.3M)|134.641s|5.5μs|25989573 (26.0M)|86.405s|3.3μs|39.8%|
|`DNSText.__eq__`|52966716 (53.0M)|190.640s|3.6μs|53604915 (53.6M)|169.104s|3.2μs|12.4%|
|`DNSService.__eq__`|52620538 (52.6M)|171.660s|3.3μs|56557448 (56.6M)|170.222s|3.0μs|7.8%|

So my conclusion from this are these:
* For `DNSEntry` my test was not conclusive. The sample size was not big enough.
* For `DNSPointer` and `DNSText` the performance was significantly better.
* For `DNSService` the performance was slightly better, on the edge of being statistically significant I'd say.